### PR TITLE
metrics: add Timer

### PR DESCRIPTION
**Pull request by peterbourgon**
As a counterproposal to #353, @brian-brazil, what do you think of this?

### CODE REVIEW

File: metrics/timer.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import "time"
+
+// Timer acts as a stopwatch, sending observations to a wrapped histogram.
+// It's a bit of helpful syntax sugar for h.Observe(time.Since(x)).
+type Timer struct {
+	h Histogram
+	t time.Time
+}
+
+// NewTimer wraps the given histogram and marks the current time.
```
Comment (brian-brazil) [inline metrics/timer.go]: I'd avoid the work "mark" as that's used by Dropwizard Meters and may cause confusion, 'records' or 'notes' should do.

File: metrics/timer_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/generic"
+)
+
+func TestTimerFast(t *testing.T) {
+	h := generic.NewSimpleHistogram()
+	metrics.NewTimer(h).ObserveDuration()
+
+	tolerance := 0.005
```
Comment (brian-brazil) [inline metrics/timer_test.go]: We've seen 10ms not be enough in timing sensitive Prometheus tests. I'd use at least 50ms.

**Comment by brian-brazil** [comment]
From a naming standpoint the Gauge will need to also support timing if batch job monitoring is added at any point.

I'm still not a fan of everyone having to import `metrics`, the approach suggested presumes that every user is happy with breaking changes anytime any of their potentially tens to hundreds of upstreams changes their metrics. I think consideration should be given to a user who wants to use just one backend, as that's a scalable approach.

File: metrics/timer_test.go (hunk 2 above)
Comment (peterbourgon) [inline metrics/timer_test.go]: For an elapsed time of zero?!

File: metrics/timer.go (hunk 1 above)
Comment (peterbourgon) [inline metrics/timer.go]: Will do!

**Comment by peterbourgon** [comment]
Thanks for the feedback!

> From a naming standpoint the Gauge will need to also support timing if batch job monitoring is added at any point.

The Timer is syntax sugar for already-supported use cases. I can _just barely_ be convinced that the Histogram use case is prolific enough to warrant expanding the surface area of the package API with a new type. But I think the Gauge use case you describe falls well short of that mark, especially considering batch jobs are rather far outside of Go kit's purview. Still, thank you!

File: metrics/timer_test.go (hunk 2 above)
Comment (ChrisHines) [inline metrics/timer_test.go]: FWIW, with Go 1.6.x on Windows, time.Now has a resolution ~15ms.

**Pull request merged**